exercises/queen-attack/test/queen_attack_test.exs

```diff
@@ -1,180 +1,212 @@
-      _ _ _ _ _ _ _ _
-      _ _ _ W _ _ _ _
-      _ _ _ _ _ _ _ _
-      _ _ _ _ _ _ _ _
-      _ _ _ _ _ _ _ B
-      """)
-
-    assert Queens.to_string(queens) == board
-  end
+    @tag :pending
+    test "can attack on same row" do
+      queens = Queens.new(white: {2, 4}, black: {2, 6})
+      assert Queens.can_attack?(queens)
+    end
 
-  @tag :pending
-  test "queen placed on the edge of the board" do
-    queens = Queens.new({4, 3}, {2, 7})
-
-    board =
-      String.trim("""
-      _ _ _ _ _ _ _ _
-      _ _ _ _ _ _ _ _
-      _ _ _ _ _ _ _ B
-      _ _ _ _ _ _ _ _
-      _ _ _ W _ _ _ _
-      _ _ _ _ _ _ _ _
-      _ _ _ _ _ _ _ _
-      _ _ _ _ _ _ _ _
-      """)
-
-    assert Queens.to_string(queens) == board
-  end
+    @tag :pending
+    test "can attack on same column" do
+      queens = Queens.new(white: {5, 4}, black: {2, 4})
+      assert Queens.can_attack?(queens)
+    end
 
-  @tag :pending
-  test "queens positioned on the same row" do
-    queens = Queens.new({4, 4}, {4, 5})
-
-    board =
-      String.trim("""
-      _ _ _ _ _ _ _ _
-      _ _ _ _ _ _ _ _
-      _ _ _ _ _ _ _ _
-      _ _ _ _ _ _ _ _
-      _ _ _ _ W B _ _
-      _ _ _ _ _ _ _ _
-      _ _ _ _ _ _ _ _
-      _ _ _ _ _ _ _ _
-      """)
-
-    assert Queens.to_string(queens) == board
-  end
+    @tag :pending
+    test "can attack on first diagonal" do
+      queens = Queens.new(white: {2, 2}, black: {0, 4})
+      assert Queens.can_attack?(queens)
+    end
 
-  @tag :pending
-  test "cannot attack" do
-    queens = Queens.new({2, 3}, {4, 7})
-    refute Queens.can_attack?(queens)
-  end
+    @tag :pending
+    test "can attack on second diagonal" do
+      queens = Queens.new(white: {2, 2}, black: {3, 1})
+      assert Queens.can_attack?(queens)
+    end
 
-  @tag :pending
-  test "can attack on same row" do
-    queens = Queens.new({2, 4}, {2, 7})
-    assert Queens.can_attack?(queens)
-  end
+    @tag :pending
+    test "can attack on third diagonal" do
+      queens = Queens.new(white: {2, 2}, black: {1, 1})
+      assert Queens.can_attack?(queens)
+    end
 
-  @tag :pending
-  test "can attack on same column" do
-    queens = Queens.new({5, 4}, {2, 4})
-    assert Queens.can_attack?(queens)
+    @tag :pending
+    test "can attack on fourth diagonal" do
+      queens = Queens.new(white: {1, 7}, black: {0, 6})
+      assert Queens.can_attack?(queens)
+    end
   end
 
   @tag :pending
-  test "can attack on diagonal" do
-    queens = Queens.new({1, 1}, {6, 6})
-    assert Queens.can_attack?(queens)
+  test "specific placement" do
+    queens = Queens.new(white: {3, 7}, black: {6, 1})
+    assert queens.white == {3, 7}
+    assert queens.black == {6, 1}
   end
 
   @tag :pending
-  test "can attack on other diagonal" do
-    queens = Queens.new({0, 6}, {1, 7})
-    assert Queens.can_attack?(queens)
+  test "cannot occupy same space" do
+    assert_raise ArgumentError, fn ->
+      Queens.new(white: {2, 4}, black: {2, 4})
+    end
```

Comment:
Could you add a test case to test these in reverse? e.g Queens.new(black: pos, white: pos)